fdbserver/ClusterController.actor.cpp

```diff
@@ -556,40 +557,50 @@ class ClusterControllerData {
 
 	RecruitFromConfigurationReply findWorkersForConfiguration( RecruitFromConfigurationRequest const& req ) {
 		if(req.configuration.regions.size() > 1) {
+			std::vector<RegionInfo> regions = req.configuration.regions;
+			if(regions[0].priority == regions[1].priority && clusterControllerDcId.present() && regions[1].dcId == clusterControllerDcId.get()) {
+				std::swap(regions[0], regions[1]);
+			}
+			if(regions[0].priority < 0) {
+				throw no_more_servers();
+			}
 			bool setPrimaryDesired = false;
 			try {
-				auto reply = findWorkersForConfiguration(req, req.configuration.regions[0].dcId);
+				auto reply = findWorkersForConfiguration(req, regions[0].dcId);
 				setPrimaryDesired = true;
 				vector<Optional<Key>> dcPriority;
-				dcPriority.push_back(req.configuration.regions[0].dcId);
-				dcPriority.push_back(req.configuration.regions[1].dcId);
+				dcPriority.push_back(regions[0].dcId);
+				dcPriority.push_back(regions[1].dcId);
 				desiredDcIds.set(dcPriority);
 				if(reply.isError()) {
 					throw reply.getError();
-				} else if(clusterControllerDcId.present() && req.configuration.regions[0].dcId == clusterControllerDcId.get()) {
+				} else if(clusterControllerDcId.present() && regions[0].dcId == clusterControllerDcId.get()) {
 					return reply.get();
 				}
 				throw no_more_servers();
 			} catch( Error& e ) {
-				if (e.code() != error_code_no_more_servers) {
+				if (e.code() != error_code_no_more_servers || regions[1].priority < 0 || now() - startTime < SERVER_KNOBS->WAIT_FOR_GOOD_RECRUITMENT_DELAY) {
```

Comment:
Man, I'd really enjoy it if `now() - startTime < ...` was a tiny inline function that had a name that described why it's done everywhere.